src/components/Wallet/WalletTransactionHistoryMobile/utils.js

```diff
@@ -0,0 +1,72 @@
+import { BN, BN_ZERO } from '@polkadot/util';
+import { formatMeritTransaction, formatDollarTransaction, formatAssetTransaction } from '../../../utils/walletHelpers';
+import formatDate from '../../../utils/formatDate';
+import paymentIcon from '../../../assets/icons/RedArrowCicrle.svg';
+import reciveIcon from '../../../assets/icons/GreenArrowCircle.svg';
+
+export const transactionHistoryProcessorFactory = (walletAddress) => (transactionHistoryInfo, index) => {
+  const value = transactionHistoryInfo.fromId === walletAddress
+    ? `-${transactionHistoryInfo.value}`
+    : transactionHistoryInfo.value;
+  const isStakingTransaction = 'userId' in transactionHistoryInfo;
```

Comment:
where did this logic come from?

Reply:
OG code from wallet transaction history for desktop